lib/sqlalchemy/dialects/postgresql/pg8000.py

```diff
@@ -658,5 +658,13 @@ def on_connect(conn):
     def _dialect_specific_select_one(self):
         return ";"
 
+    def do_terminate(self, dbapi_connection) -> None:
+        try:
+            dbapi_connection.close()
+        except self.dbapi.InterfaceError as e:
+            msg = str(e)
+            if not ("network error" in msg or "connection is closed" in msg):
+                raise e
+
```

Comment:
does pg8000 document / guarantee that if we arrive inside this block, the client connection has discarded all of its network state ?       I continue to be a little uncomfortable with the approach here.  I mean, we could use this do_terminate() for every DBAPI if we are just assuming that "InterfaceError" means "we definitely discarded the connection to the best of our ability"

this is why do_terminate() was intended to accommodate a real "terminate()" method on each DBAPI